bigframes/core/compile/sqlglot/expressions/unary_compiler.py

```diff
@@ -111,11 +147,88 @@ def _(op: ops.ArraySliceOp, expr: TypedExpr) -> sge.Expression:
     return sge.array(selected_elements)
 
 
+@UNARY_OP_REGISTRATION.register(ops.capitalize_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    return sge.Initcap(this=expr.expr)
+
+
+@UNARY_OP_REGISTRATION.register(ops.ceil_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    return sge.Ceil(this=expr.expr)
+
+
 @UNARY_OP_REGISTRATION.register(ops.cos_op)
 def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
     return sge.func("COS", expr.expr)
 
 
+@UNARY_OP_REGISTRATION.register(ops.cosh_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    return sge.Case(
+        ifs=[
+            sge.If(
+                this=sge.func("ABS", expr.expr) > sge.convert(709.78),
+                true=_INF,
+            )
+        ],
+        default=sge.func("COSH", expr.expr),
+    )
+
+
+@UNARY_OP_REGISTRATION.register(ops.date_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    return sge.Date(this=expr.expr)
+
+
+@UNARY_OP_REGISTRATION.register(ops.day_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    return sge.Extract(this=sge.Identifier(this="DAY"), expression=expr.expr)
+
+
+@UNARY_OP_REGISTRATION.register(ops.dayofweek_op)
+def _(op: ops.base_ops.UnaryOp, expr: TypedExpr) -> sge.Expression:
+    # Plus 1 because iso day of week uses 1-based indexing
+    return sge.Extract(
+        this=sge.Identifier(this="DAYOFWEEK"), expression=expr.expr
+    ) + sge.convert(1)
```

Comment:
I think it should be minus 1 because SQL uses 1-based indexing while Pandas uses 0-based